### PR TITLE
Dhanush/update payment api

### PR DESCRIPTION
**Pull request by dhanush-hetulabs**

### CODE REVIEW

File: controllers/payments_controller.js [hunk 1]
```diff
@@ -16,4 +16,18 @@ exports.create_payment = async(req,resp)=>{
         return resp.status(status_code).send({ status: "Failure", message: "payment not created" })
     }finally{
     }
+}
+
+exports.update_payment = async(req,resp)=>{
+
```
Comment (ssi259) [inline controllers/payments_controller.js:22]: remove space

File: routes/payment.js [hunk 2]
```diff
@@ -5,4 +5,5 @@ const {auth} = require('../middlewares/authentication')
 
 
 router.post('/', auth, payment_controller.create_payment);
+router.post('/update', auth, payment_controller.update_payment);
```
Comment (ssi259) [inline routes/payment.js:8]: has to  be PUT API.endpoint has  to be /

File: services/payment_manager.js [hunk 3]
```diff
@@ -29,7 +35,7 @@ exports.process_create_input_req = async(req,input_response)=>{
     await models.Payment.update({status: "failed"}, {where: {user_id: req.user.user_id,plan_id: req.body.plan_id,batch_id:req.body.batch_id}})
   } 
   
-    await models.Payment.create({plan_id: req.body.plan_id,status: "pending",user_id: req.user.user_id,coach_id: req.body.coach_id})
+    await models.Payment.create({plan_id: req.body.plan_id,status: "pending",user_id: req.user.user_id,coach_id: req.body.coach_id, price: plan.price, payment_made: "Cash"})
```
Comment (ssi259) [inline services/payment_manager.js:38]: correct this to  payment_mode

File: services/payment_manager.js [hunk 4]
```diff
@@ -42,5 +48,62 @@ exports.post_create_process = async(req,resp,input_response)=>{
...
+  }
+})
+
+return {payment_data, enrollment_data, plan, coach_resp}
+}
+
+exports.process_update_input_req = async(req,input_response)=>{
+
+  const {payment_data, enrollment_data, plan, coach_resp} = input_response;
+
```
Comment (ssi259) [inline services/payment_manager.js:84]: remove extra spaces

File: services/payment_manager.js [hunk 5]
```diff
@@ -42,5 +48,62 @@ exports.post_create_process = async(req,resp,input_response)=>{
   delete input_response.dataValues.user_id
   formatted_response["data"]=input_response
 
+  resp.status(200).send(formatted_response)
+}
+
+exports.pre_process_update = async(req,resp)=>{
+
```
Comment (ssi259) [inline services/payment_manager.js:55]: correct   indentation and remove spaces

File: controllers/payments_controller.js (hunk 1 above)
Comment (aalesh01-hetu) [reply controllers/payments_controller.js:22]: remove spaces here

File: services/payment_manager.js (hunk 5 above)
Comment (aalesh01-hetu) [reply services/payment_manager.js:54]: please remove blank lines

**Pull request merged**